heap_min.go

```diff
@@ -2,32 +2,57 @@ package ch
 
 import "container/heap"
 
-type minHeapVertex struct {
-	id       int64
-	distance float64
+type MinHeapVertex struct {
+	ID       int64
+	Distance float64
 }
 
-type minHeap []*minHeapVertex
+type MinHeap struct {
+	ids       []int64
+	distances map[int64]float64
+}
+
+func NewMinHeap() *MinHeap {
+	return &MinHeap{
+		ids:       make([]int64, 0),
```

Comment:
Is there a reason to use `make` here instead of `nil`?

Reply:
Is there a reason to use nil here instead of make? :)

Reply:
`nil` is more [idiomatic](https://go.dev/wiki/CodeReviewComments#declaring-empty-slices).